File: revisi/app/src/main/java/com/ukom/resto2/ui/dailymeal/DailyMealFragment.java
package com.ukom.resto2.ui.dailymeal;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.ukom.resto2.R;
import com.ukom.resto2.adapters.DailyMealAdapter;
import com.ukom.resto2.models.DailyMealModel;

import java.util.ArrayList;
import java.util.List;

public class DailyMealFragment extends Fragment {

    RecyclerView recyclerView;
    List<DailyMealModel> dailyMealModels;
    DailyMealAdapter dailyMealAdapter;
    ImageView imageView;

    public View onCreateView(@NonNull LayoutInflater inflater,
                             ViewGroup container, Bundle savedInstanceState) {
        View root = inflater.inflate(R.layout.daily_meal_fragment, container, false);

        recyclerView = root.findViewById(R.id.daily_meal_rec);
        recyclerView.setLayoutManager(new LinearLayoutManager(getContext()));
        dailyMealModels = new ArrayList<>();

        dailyMealModels.add(new DailyMealModel(R.drawable.breakfast,"Breakfast","30% OFF","Description","breakfast"));
        dailyMealModels.add(new DailyMealModel(R.drawable.lunch,"Lunch","30% OFF","Description","lunch"));
        dailyMealModels.add(new DailyMealModel(R.drawable.dinner,"Dinner","20% OFF","Description","dinner"));
        dailyMealModels.add(new DailyMealModel(R.drawable.sweets,"Sweets","39% OFF","Description","sweets"));
        dailyMealModels.add(new DailyMealModel(R.drawable.coffe,"Coffee","15% OFF","Description","coffee"));

        dailyMealAdapter = new DailyMealAdapter(getContext(),dailyMealModels);
        recyclerView.setAdapter(dailyMealAdapter);
        dailyMealAdapter.notifyDataSetChanged();
        return root;
    }
}

File: revisi/app/src/main/java/com/ukom/resto2/ui/dailymeal/DailyMealFragment.kt
package com.ukom.resto2.ui.dailymeal

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.ukom.resto2.R
import com.ukom.resto2.adapters.DailyMealAdapter
import com.ukom.resto2.models.DailyMealModel

class DailyMealFragment : Fragment() {
    var recyclerView: RecyclerView? = null
    var dailyMealModels: MutableList<DailyMealModel>? = null
    var dailyMealAdapter: DailyMealAdapter? = null
    var imageView: ImageView? = null

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?, savedInstanceState: Bundle?
    ): View? {
        val root = inflater.inflate(R.layout.daily_meal_fragment, container, false)

        recyclerView = root.findViewById(R.id.daily_meal_rec)
        recyclerView?.setLayoutManager(LinearLayoutManager(context))
        dailyMealModels = ArrayList()

        (dailyMealModels as ArrayList<DailyMealModel>).add(
            DailyMealModel(
                R.drawable.breakfast,
                "Breakfast",
                "30% OFF",
                "Description",
                "breakfast"
            )
        )
        (dailyMealModels as ArrayList<DailyMealModel>).add(
            DailyMealModel(
                R.drawable.lunch,
                "Lunch",
                "30% OFF",
                "Description",
                "lunch"
            )
        )
        (dailyMealModels as ArrayList<DailyMealModel>).add(
            DailyMealModel(
                R.drawable.dinner,
                "Dinner",
                "20% OFF",
                "Description",
                "dinner"
            )
        )
        (dailyMealModels as ArrayList<DailyMealModel>).add(
            DailyMealModel(
                R.drawable.sweets,
                "Sweets",
                "39% OFF",
                "Description",
                "sweets"
            )
        )
        (dailyMealModels as ArrayList<DailyMealModel>).add(
            DailyMealModel(
                R.drawable.coffe,
                "Coffee",
                "15% OFF",
                "Description",
                "coffee"
            )
        )

        dailyMealAdapter = context?.let { DailyMealAdapter(it, dailyMealModels as ArrayList<DailyMealModel>) }
        recyclerView?.setAdapter(dailyMealAdapter)
        dailyMealAdapter!!.notifyDataSetChanged()
        return root
    }
}